compile error with clang-22 on debian 11

I'm trying compile simdutf on my debian 11 comptuer, with clang-22 from https://apt.llvm.org

```
gillesvollant@gv-w11x64:/mnt/w/devw$ clang++-22 -std=c++23 -o vir exterieur/simdutf/singleheader/simdutf.cpp
In file included from exterieur/simdutf/singleheader/simdutf.cpp:3:
exterieur/simdutf/singleheader/simdutf.h:14111:10: error: no member named 'unreachable' in namespace 'std'
 14111 |     std::unreachable(); // invalid base64 input in _base64 literal
       |          ^~~~~~~~~~~
exterieur/simdutf/singleheader/simdutf.cpp:33262:32: warning: array index 2 is past the end of the array (that has type 'const simd8<unsigned char>[2]') [-Warray-bounds]
 33262 |         this->check_utf8_bytes(input.chunks[2], input.chunks[1]);
       |                                ^            ~
exterieur/simdutf/singleheader/simdutf.cpp:4826:3: note: array 'chunks' declared here
 4826 |   simd8<T> chunks[NUM_CHUNKS];
      |   ^
exterieur/simdutf/singleheader/simdutf.cpp:33263:32: warning: array index 3 is past the end of the array (that has type 'const simd8<unsigned char>[2]') [-Warray-bounds]
 33263 |         this->check_utf8_bytes(input.chunks[3], input.chunks[2]);
       |                                ^            ~
exterieur/simdutf/singleheader/simdutf.cpp:4826:3: note: array 'chunks' declared here
 4826 |   simd8<T> chunks[NUM_CHUNKS];
      |   ^
exterieur/simdutf/singleheader/simdutf.cpp:33263:49: warning: array index 2 is past the end of the array (that has type 'const simd8<unsigned char>[2]') [-Warray-bounds]
 33263 |         this->check_utf8_bytes(input.chunks[3], input.chunks[2]);
       |                                                 ^            ~
exterieur/simdutf/singleheader/simdutf.cpp:4826:3: note: array 'chunks' declared here
 4826 |   simd8<T> chunks[NUM_CHUNKS];
      |   ^
3 warnings and 1 error generated.
```

I've no error with  --std=c++20

--- a/src/generic/utf8_to_latin1/utf8_to_latin1.h
+++ b/src/generic/utf8_to_latin1/utf8_to_latin1.h
@@ -144,10 +144,10 @@ struct validating_transcoder {
                 (simd8x64<uint8_t>::NUM_CHUNKS == 4),
             "We support either two or four chunks per 64-byte block.");
         auto zero = simd8<uint8_t>{uint8_t(0)};
-        if (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
+        if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
-        } else if (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
+        } else if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
           this->check_utf8_bytes(input.chunks[2], input.chunks[1]);
@@ -233,10 +233,10 @@ struct validating_transcoder {
                 (simd8x64<uint8_t>::NUM_CHUNKS == 4),
             "We support either two or four chunks per 64-byte block.");
         auto zero = simd8<uint8_t>{uint8_t(0)};
-        if (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
+        if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
-        } else if (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
+        } else if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
           this->check_utf8_bytes(input.chunks[2], input.chunks[1]);

--- a/src/generic/utf8_to_utf32/utf8_to_utf32.h
+++ b/src/generic/utf8_to_utf32/utf8_to_utf32.h
@@ -157,10 +157,10 @@ struct validating_transcoder {
                 (simd8x64<uint8_t>::NUM_CHUNKS == 4),
             "We support either two or four chunks per 64-byte block.");
         auto zero = simd8<uint8_t>{uint8_t(0)};
-        if (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
+        if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
-        } else if (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
+        } else if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
           this->check_utf8_bytes(input.chunks[2], input.chunks[1]);
@@ -244,10 +244,10 @@ struct validating_transcoder {
                 (simd8x64<uint8_t>::NUM_CHUNKS == 4),
             "We support either two or four chunks per 64-byte block.");
         auto zero = simd8<uint8_t>{uint8_t(0)};
-        if (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
+        if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
-        } else if (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
+        } else if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
           this->check_utf8_bytes(input.chunks[0], zero);
           this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
           this->check_utf8_bytes(input.chunks[2], input.chunks[1]);

--- a/src/generic/utf8_validation/utf8_lookup4_algorithm.h
+++ b/src/generic/utf8_validation/utf8_lookup4_algorithm.h
@@ -193,10 +193,10 @@ struct utf8_checker {
       static_assert((simd8x64<uint8_t>::NUM_CHUNKS == 2) ||
                         (simd8x64<uint8_t>::NUM_CHUNKS == 4),
                     "We support either two or four chunks per 64-byte block.");
-      if (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
+      if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 2) {
         this->check_utf8_bytes(input.chunks[0], this->prev_input_block);
         this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
-      } else if (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
+      } else if constexpr (simd8x64<uint8_t>::NUM_CHUNKS == 4) {
         this->check_utf8_bytes(input.chunks[0], this->prev_input_block);
         this->check_utf8_bytes(input.chunks[1], input.chunks[0]);
         this->check_utf8_bytes(input.chunks[2], input.chunks[1]);

